### PR TITLE
Feature/legacy sdk cleanup

### PR DESCRIPTION
**Pull request by alberttorosyan**
* Remove legacy AimRepo (keep only stuff required for repo convert CLI) 
* Remove legacy CLI commands (aim expriment, etc.) update remaining ones to use new Repo 
* Update aim.sdk.Session class to use new Repo 
* aim.storage enhancements (run step auto-increment, run hash generation, repo path handling, etc. )

### CODE REVIEW

File: aim/sdk/__init__.py [hunk 1]
```diff
@@ -1,18 +1,4 @@
 # Main functions
 from aim.sdk.track import track, set_params
 from aim.sdk.flush import flush
-from aim.sdk.init import init
```
Comment (gorarakelyan) [inline aim/sdk/__init__.py:4]: `init` method should be kept, it used to create a default session for forwarding `aim.track` to `default_session.track`.

Comment (gorarakelyan) [reply aim/sdk/__init__.py:4]: It was also used for specifying experiment name and repo path of default session.

File: aim/sdk/session/session.py [hunk 2]
```diff
@@ -2,79 +2,42 @@
...
-    Job = namedtuple('Job', 'target, args, kwargs')
+class Session:
+    sessions = {}
 
     _are_exit_listeners_set = False
     _original_sigint_handler = None
     _original_sigterm_handler = None
 
     def __init__(self, repo: Optional[str] = None,
                  experiment: Optional[str] = None,
```
Comment (gorarakelyan) [inline aim/sdk/session/session.py:24]: Add deprecation warning.

File: aim/engine/utils.py [hunk 3]
```diff
@@ -1,311 +0,0 @@
...
-    if is_numpy_number(value):
-        return True
-
-    if is_pytorch_tensor(value):
-        return True
-
-    return False
-
-
-def convert_to_py_number(value):
```
Comment (gorarakelyan) [inline aim/engine/utils.py:247]: if I am not wrong, this function is used in `sdk/adapters` package.

**Review by gorarakelyan — COMMENTED** [review]
what do you think about moving `aim/storage/sdk` to `aim/sdk`?

**Comment by gorarakelyan** [comment]
The rest seems good! 👍

**Comment by gorarakelyan** [comment]
+ resource tracker should be initiated in `Run` as well, it tracks system resource usage

File: aim/sdk/__init__.py [hunk 4]
```diff
@@ -1,18 +1,4 @@
...
-from aim.sdk.select import select_metrics, select_runs
```
Comment (gorarakelyan) [inline aim/sdk/__init__.py:5]: forward to runs and metrics new search interfaces? +add deprecation warnings

File: aim/sdk/session/session.py (hunk 2 above)
Comment (alberttorosyan) [reply aim/sdk/session/session.py:24]: sure

File: aim/engine/utils.py (hunk 3 above)
Comment (alberttorosyan) [reply aim/engine/utils.py:247]: will double-check this one

File: aim/sdk/__init__.py (hunk 1 above)
Comment (alberttorosyan) [reply aim/sdk/__init__.py:4]: should we stub `init` method or should we keep this behavior? as of now, `Repo` (and legacy `Session`) do not require special initialization before usage.  the only thing required is `aim init` being ran for that repo.

File: aim/sdk/__init__.py (hunk 4 above)
Comment (alberttorosyan) [reply aim/sdk/__init__.py:5]: > what do you think about moving `aim/storage/sdk` to `aim/sdk`?

totally makes sense. but only after current `aim/sdk` contents are finally dropped.

File: examples/keras_track.py [hunk 5]
```diff
@@ -1,6 +1,6 @@
 from aim.keras import AimCallback
 
-import keras
+from tensorflow import keras
```
Comment (gorarakelyan) [inline examples/keras_track.py:3]: This line should remain as `import keras`, because before tf2 keras was a separate high-level ML framework.

**Review by gorarakelyan — APPROVED** [review]
looks great

**Review by mihran113 — APPROVED** [review]
fantastic job!
https://github.com/aimhubio/aim/pull/603#discussion_r692867563
just noticed you didn't address this comment by Gor, everything else is good.

File: examples/keras_track.py (hunk 5 above)
Comment (alberttorosyan) [reply examples/keras_track.py:3]: 👌

**Pull request merged**